src/datalad_installer.py

```diff
@@ -1454,6 +1454,12 @@ class DebURLInstaller(Installer):
 
     OPTIONS = [
         Option("--url", metavar="URL", help="URL from which to download `*.deb` file"),
+        Option(
+            "--install-dir",
+            converter=Path,
+            metavar="DIR",
+            help="Directory in which to unpack the `*.deb`",
+        ),
```

Comment:
Sorry it took me too long to get to this PR -- I for some reason even thought we had it merged already ;-)

for my typical use-case (quickly test a fresh build on a system where I can't `dpkg -i` it), I would prefer to not bother specifying target directory since ideally it should have annex version which I do not know. 

Would it be possible to make it handle e.g. empty value for a directory it would place it under the temp dir  along side with the temp env file?

<details>
<summary>Blindly trying it while giving an empty DIR</summary> 

```shell
(git)lena:~datalad/datalad-installer[gh-71]git
$> datalad-installer git-annex --install-dir '' -m datalad/git-annex
2021-07-07T14:29:01-0400 [INFO    ] datalad_installer Writing environment modifications to /home/yoh/.tmp/dl-env-uidizwfv.sh
2021-07-07T14:29:01-0400 [INFO    ] datalad_installer Installing git-annex via datalad/git-annex
2021-07-07T14:29:01-0400 [INFO    ] datalad_installer Getting artifacts_url from https://api.github.com/repos/datalad/git-annex/actions/workflows/build-ubuntu.yaml/runs?status=success&branch=master
2021-07-07T14:29:04-0400 [INFO    ] datalad_installer Getting archive download URL from https://api.github.com/repos/datalad/git-annex/actions/runs/873457349/artifacts
2021-07-07T14:29:05-0400 [INFO    ] datalad_installer Downloading artifact package from https://api.github.com/repos/datalad/git-annex/actions/artifacts/62821844/zip
2021-07-07T14:29:05-0400 [INFO    ] datalad_installer Downloading https://api.github.com/repos/datalad/git-annex/actions/artifacts/62821844/zip
2021-07-07T14:31:23-0400 [INFO    ] datalad_installer Running: dpkg -x '/home/yoh/.tmp/tmp2akpv3dy/git-annex-standalone_8.20210428+git202-g9a5981a15-1~ndall+1_amd64.deb' .
2021-07-07T14:31:24-0400 [INFO    ] datalad_installer git-annex is now installed at usr/bin/git-annex
(dev3) 1 20390.....................................:Wed 07 Jul 2021 02:31:24 PM EDT:.
(git)lena:~datalad/datalad-installer[gh-71]git
$> ls usr/bin/git-annex
usr/bin/git-annex@
(dev3) 1 20391.....................................:Wed 07 Jul 2021 02:31:31 PM EDT:.
(git)lena:~datalad/datalad-installer[gh-71]git
$> ls usr/bin/git-annex -l
lrwxrwxrwx 1 yoh yoh 32 May 24 23:25 usr/bin/git-annex -> ../lib/git-annex.linux/git-annex*

$> cat /home/yoh/.tmp/dl-env-uidizwfv.sh
export PATH=usr/bin:"$PATH"

```
</details>

placed it under current directory and also added that relative path `usr/bin` to PATH -- IMHO it should ensure that path is absolute before adding anything to the PATH

Reply:
Do you want the folder name to include the git-annex version?

Reply:
@yarikoptic More importantly, if `--install-dir` isn't specified, how should the script tell whether to install git-annex system wide or in a temporary directory?

Reply:
@yarikoptic How about this: If the `--install-dir` includes a `{version}` placeholder, the script will fill it in with the git-annex version.

Reply:
having `{version}` sounds great! Just may be add also `{tmpdir}` (datalad has that e.g. for `run` command etc) so I could "install" throw away instances with `--install-dir {tmpdir}/{version}` ?

Reply:
@yarikoptic Do you want `{tmpdir}` to be supported anywhere else besides deb-url's `--install-dir`?

Reply:
I don't have an immediate usecase, but if easy and would make it more consistent, why not?

Reply:
I've added `{tmpdir}` support to `deb-url`'s `--install-dir` and `datalad/git-annex`'s `--install-dir`.  It doesn't really make sense for other path options, as they already default to tmpdirs when not specified.

Reply:
note: and `{version}` only where known, so not for e.g. `datalad/git-annex*` ... `{tmpdir}` seems to work. Thank you!